### PR TITLE
Add test plan folder structure, template and readme

### PR DESCRIPTION
**Pull request by cdingm**
Closes Gamify-IT/issues#191
there is now a folder structure i came up with
a readme how this should be used
and a translated template of the testplan

feel free to discuss if this structure makes sense, and if the readme is understandable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,9 +24,9 @@ Contains the user manual for the given repo, so how the user can start the given
 
 Contains protocols of meetings regarding the given repo/ concept for the given day. Filenames are in the format `YYYY-mm-dd-protocol.md`.
 
-### [`/internal`](internal/README.md)
+### [`/test-plans`](test-plans)
```
Comment (delvh) [inline README.md:27]: See my comment on the issue.
I think it makes the most sense to see them as part of the dev-manuals, that's where they are most logical.

File: test-plans/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# test plans
```
Comment (delvh) [inline test-plans/README.md:1]:
```suggestion
# Test plans
```

File: test-plans/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Contains all the test-plans of all our services, and the test plan templates.
```
Comment (delvh) [inline test-plans/README.md:3]:
```suggestion
Contains all test plans of all our services, and the templates (warning: files inside this folder are in German).
```

File: test-plans/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## folder structure
```
Comment (delvh) [inline test-plans/README.md:5]:
```suggestion
## Folder structure
```

File: test-plans/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+the test-plans belong to their corresponding folder. For Example a test plan created for the Moorhuhn frontend:
```
Comment (delvh) [inline test-plans/README.md:7]:
```suggestion
the test plans belong to their corresponding folder. For example, a test plan created for the Moorhuhn frontend:
```

File: test-plans/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+test-plans/minigames/moorhuhn/frontend/plans
+```
+
+As soon as a tester has tested a feature according to a test plan, it has to be placed in the `executed` folder
```
Comment (delvh) [inline test-plans/README.md:13]:
```suggestion
As soon as a tester has tested a feature according to a test plan, it has to be placed in the `test-log` folder
```

File: test-plans/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## naming convention
```
Comment (delvh) [inline test-plans/README.md:15]:
```suggestion
## Naming convention
```

File: test-plans/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+test-plan files should be following this structure
```
Comment (delvh) [inline test-plans/README.md:17]:
```suggestion
Test plan filenames should have the following structure:
```

File: test-plans/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# test plans
+
+Contains all the test-plans of all our services, and the test plan templates.
+
+## folder structure
+
+the test-plans belong to their corresponding folder. For Example a test plan created for the Moorhuhn frontend:
+
+```
+test-plans/minigames/moorhuhn/frontend/plans
```
Comment (delvh) [inline test-plans/README.md:10]:
```suggestion
```bash
test-plans/services/moorhuhn/plans.[md,pdf]
```

File: test-plans/README.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+As soon as a tester has tested a feature according to a test plan, it has to be placed in the `executed` folder
+
+## naming convention
+
+test-plan files should be following this structure
+
+```
+name-service-backend/frontend.md
+example:
+highscore-moorhuhn-frontend.md
```
Comment (delvh) [inline test-plans/README.md:22]:
````suggestion
```bash
$ACTION-$SERVICE.[md,pdf]
example:
highscore-moorhuhn-frontend.md
````

File: test-plans/README.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+where `name` is the name of the feature the test plan covers
```
Comment (delvh) [inline test-plans/README.md:25]:
```suggestion
where `$ACTION` is the name of the feature the test plan covers
```
↑ Also, as I understood it: There should basically be one test plan per service, each having multiple test cases.

File: test-plans/README.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+executed tests should be named
+
+```
+date-testername-name-service-backend/frontend.md
+example:
+2022-08-01-MichaelLinder-highscore-moorhuhn-frontend.md
```
Comment (delvh) [inline test-plans/README.md:32]:
```suggestion
```bash
$DATE-$TESTER-$ACTION_NAME-$SERVICE_NAME-testresults.md
example:
2022-08-01-michael_linder-highscore-moorhuhn-frontend.md
```

File: test-plans/template.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
+# Name
```
Comment (delvh) [inline test-plans/template.md:1]: Don't we use @LaviniaStiliadou's PDF template?

File: test-plans/template.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+
+Number: Unique and unambiguous identifier according to the following
+pattern:
+U.XX(service identifier).X (sequential number, start
+at 1).  
+Version: Version of the use case according to pattern.
```
Comment (delvh) [inline test-plans/template.md:7]: I'd simply use the next planned version of the tested service...

**Comment by cdingm** [comment]
pdfs in github mmhhhh i dont like it, but well talk about it in the daily

File: test-plans/template.md (hunk 14 above)
Comment (cdingm) [reply test-plans/template.md:7]: i copied the descriptions of cheng from teams

File: test-plans/README.md (hunk 3 above)
Comment (cdingm) [reply test-plans/README.md:3]: should we do it in german? we should discuss this in the daily

File: test-plans/template.md (hunk 13 above)
Comment (cdingm) [reply test-plans/template.md:1]: no

File: test-plans/README.md (hunk 3 above)
Comment (delvh) [reply test-plans/README.md:3]: As discussed:
Luckily, the majority is for English, so our test plans will also be in English.

File: dev-manuals/test-plans/README.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
+# Test plans
+
+Contains all the test-plans of all our services, and the test plan templates.
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:3]:
```suggestion
Contains all test plans of all our services, and the test plan template.
```

File: dev-manuals/test-plans/README.md [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+
+## Folder structure
+
+the test plans belong to their corresponding folder. For example, a test plan created for the Moorhuhn frontend:
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:7]:
```suggestion
The test plans belong to their corresponding folder.  
For example, a test plan created for the Moorhuhn frontend:
```

File: dev-manuals/test-plans/README.md [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+test-plans/minigames/moorhuhn/frontend/plans
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:10]:
```suggestion
```bash
test-plans/moorhuhn-frontend/plans
```

File: dev-manuals/test-plans/README.md [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+As soon as a tester has tested a feature according to a test plan, it has to be placed in the `executed` folder
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:13]:
```suggestion
As soon as a tester has tested a feature according to a test plan, it has to be placed in the `logs` subfolder of that service.
```

File: dev-manuals/test-plans/README.md [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+
+## Naming convention
+
+Test plan filenames should have the following structure:
+
+```
+name-service-backend/frontend.md
+example:
+highscore-moorhuhn-frontend.md
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:22]:
````suggestion
```bash
$ACTION-$SERVICE.md
```
So, for example
```bash
highscore-moorhuhn-frontend.md
````

File: dev-manuals/test-plans/README.md [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+where `name` is the name of the feature the test plan covers
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:25]:
```suggestion
where `$ACTION` is the name of the feature the test plan covers.
```

File: dev-manuals/test-plans/README.md [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+
+executed tests should be named
+
+```
+date-testername-name-service-backend/frontend.md
+example:
+2022-08-01-MichaelLinder-highscore-moorhuhn-frontend.md
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:32]:
````suggestion
Test log filename format is
```bash
$DATE-$TESTER-$ACTION-$SERVICE.md
```
So, for example
```bash
2022-08-01-michael_linder-highscore-moorhuhn-frontend.md
````

File: dev-manuals/test-plans/README.md [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+name-service-backend/frontend.md
+example:
+highscore-moorhuhn-frontend.md
+```
+
+where `name` is the name of the feature the test plan covers
+
+executed tests should be named
```
Comment (delvh) [inline dev-manuals/test-plans/README.md:27]:
```suggestion
Executed tests should be named
```

**Review by delvh — COMMENTED** [review]
Regarding the `.gitkeep`s: I think they are currently at the wrong location:
I would suggest not separating in front- and backend as a separate folder level, but to instead split the services into those parts, so that we have i.e. `dev-manuals/test-plans/lecturer-interface-backend/plans/.gitkeep` and ``dev-manuals/test-plans/lecturer-interface-frontend/plans/.gitkeep`` instead of `dev-manuals/test-plans/lecturer-interface/backend/plans/.gitkeep` and `dev-manuals/test-plans/lecturer-interface/frontendend/plans/.gitkeep`.
``

**Comment by cdingm** [comment]
> Regarding the `.gitkeep`s: I think they are currently at the wrong location: I would suggest not separating in front- and backend as a separate folder level, but to instead split the services into those parts, so that we have i.e. `dev-manuals/test-plans/lecturer-interface-backend/plans/.gitkeep` and `dev-manuals/test-plans/lecturer-interface-frontend/plans/.gitkeep` instead of `dev-manuals/test-plans/lecturer-interface/backend/plans/.gitkeep` and `dev-manuals/test-plans/lecturer-interface/frontend/plans/.gitkeep`.

done

**Comment by cdingm** [comment]
what did your commits do? wasnt that what i did?

**Review by delvh — APPROVED** [review]
LGTM from me.
Do we wait for other reviewers, or can we merge now?

**Review by WuestFlorian — APPROVED** [review]

**Pull request merged**